graph/src/blockchain/mod.rs

```diff
@@ -208,6 +205,8 @@ pub trait Blockchain: Debug + Sized + Send + Sync + Unpin + 'static {
     fn runtime_adapter(&self) -> Arc<dyn RuntimeAdapter<Self>>;
 
     fn chain_client(&self) -> Arc<ChainClient<Self>>;
+
+    fn block_ingestor(&self) -> anyhow::Result<Box<dyn BlockIngestor>>;
```

Comment:
Arweave, NEAR, and Cosmos pretty much have identical implementations. Unfortunately `Self::Block: prost::Message` is not guaranteed so we can't have a default implementation, among other reasons, but a helper function might be nice.

Reply:
True, I think this is prolly a good further improvement for a different PR though

Reply:
https://github.com/graphprotocol/graph-node/issues/4418

Reply:
:+1: